addon/services/unified-event-handler.js

```diff
@@ -145,6 +148,24 @@ export default Ember.Service.extend(Ember.Evented, {
     this.on(domHandlerInfo.emberEventName, callback);
   },
 
+  willDestroy() {
+    let handlerMap = this[_HANDLER_MAP];
+
+    for (let target in handlerMap) {
+      if (handlerMap.hasOwnProperty(target)) {
```

Comment:
With the change above to `Object.create(null)`, we shouldn't need this check anymore.

Can we do a similar thing for the nested objects as well?